pkg/resources/function_commons.go

```diff
@@ -0,0 +1,341 @@
+		Type:        schema.TypeString,
+		Required:    true,
+		ForceNew:    true,
+		Description: "Specifies the results returned by the UDF, which determines the UDF type. Use `<result_data_type>` to create a scalar UDF that returns a single value with the specified data type. Use `TABLE (col_name col_data_type, ...)` to creates a table UDF that returns tabular results with the specified table column(s) and column type(s). For the details, consult the [docs](https://docs.snowflake.com/en/sql-reference/sql/create-function#all-languages).",
+		// TODO [SNOW-1348103]: adjust DiffSuppressFunc
+	},
+	"null_input_behavior": {
+		Type:             schema.TypeString,
+		Optional:         true,
+		ForceNew:         true,
+		ValidateDiagFunc: sdkValidation(sdk.ToNullInputBehavior),
+		DiffSuppressFunc: SuppressIfAny(NormalizeAndCompare(sdk.ToNullInputBehavior), IgnoreChangeToCurrentSnowflakeValueInShow("null_input_behavior")),
+		Description:      fmt.Sprintf("Specifies the behavior of the function when called with null inputs. Valid values are (case-insensitive): %s.", possibleValuesListed(sdk.AllAllowedNullInputBehaviors)),
+	},
+	"return_behavior": {
+		Type:             schema.TypeString,
+		Optional:         true,
+		ForceNew:         true,
+		ValidateDiagFunc: sdkValidation(sdk.ToReturnResultsBehavior),
+		DiffSuppressFunc: SuppressIfAny(NormalizeAndCompare(sdk.ToReturnResultsBehavior), IgnoreChangeToCurrentSnowflakeValueInShow("return_behavior")),
+		Description:      fmt.Sprintf("Specifies the behavior of the function when returning results. Valid values are (case-insensitive): %s.", possibleValuesListed(sdk.AllAllowedReturnResultsBehaviors)),
+	},
+	"runtime_version": {
+		Type:     schema.TypeString,
+		Optional: true,
+		ForceNew: true,
+		// TODO [SNOW-1348103]: may be optional for java without consequence because if it is not set, the describe is not returning any version.
+	},
+	"comment": {
+		Type:     schema.TypeString,
+		Optional: true,
+		// TODO [SNOW-1348103]: handle dynamic comment - this is a workaround for now
+		Default:     "user-defined function",
+		Description: "Specifies a comment for the function.",
+	},
+	// TODO [SNOW-1348103]: because of https://docs.snowflake.com/en/sql-reference/sql/create-function#id6, maybe it will be better to split into stage_name + target_path
+	"imports": {
+		Type:     schema.TypeSet,
+		Elem:     &schema.Schema{Type: schema.TypeString},
+		Optional: true,
+		ForceNew: true,
+	},
+	// TODO [SNOW-1348103]: what do we do with the version "latest".
+	"packages": {
+		Type:     schema.TypeSet,
+		Elem:     &schema.Schema{Type: schema.TypeString},
+		Optional: true,
+		ForceNew: true,
+	},
+	"handler": {
+		Type:     schema.TypeString,
+		Required: true,
+		ForceNew: true,
+	},
+	"external_access_integrations": {
+		Type: schema.TypeSet,
+		Elem: &schema.Schema{
+			Type:             schema.TypeString,
+			ValidateDiagFunc: IsValidIdentifier[sdk.AccountObjectIdentifier](),
+		},
+		Optional:    true,
```

Comment:
Diff suppress function used e.g. in network policies can be used here

Reply:
I will add this with the tests in the next PRs, I did not want to overuse the uncommon alidations/suppressions without tests

Reply:
Added a TODO